### ISSUE TITLE
Webconf loads whole instance of zynthian-ui which is bloated and can cause issues

### ISSUE THREAD

**Issue by riban-bw** [zynthian/zynthian-issue-tracking#479]
**Describe the bug**
When webconf starts it imports some zynthian-ui modules which triggers the initialisation of all zyngui modules. This adds to the  bloat of webconf but more significantly impacts design and operation of zynthian. Modules need to be written in such a way that their re-initialisation does not interfer with the core application. This is a support burden. There is the possibility that webconf may trigger behaviour that confuses the UI or corrupts data.

**Setup:**
HARDWARE
Raspberry Pi 4 Model B Rev 1.4
Soundcard: HifiBerry DAC+ ADC PRO
Display: ZynScreen 3.5 (v1)
Wiring: MCP23017_ZynScreen
GPIO Expander: MCP23017

SYSTEM
Raspbian GNU/Linux 10 (buster)
Build Date: 2020-09-05
Memory: 58% (1129M/1939M)
SD Card: 38% (11G/29G)
Temperature: 49.0ºC

MIDI
Profile: default.sh
Fine Tuning: 440 Hz
Master Channel: 16
Single Active Channel: off
ZS3 SubSnapShots: on

SOFTWARE
zyncoder: testing (26ffaed)
zynthian-ui: bug/477_xruns (a65b997)
zynthian-sys: testing (36e814a)
zynthian-data: testing (315390c)
zynthian-webconf: testing (d09f0ba)

**Additional context**
This was observed because webconf is initialising an instance of zynseq library (and others) which triggered an issue. The underlying issue is now resolved in the library and UI code but webconf continues to load these libraries that it does not need. This could be increasing processing load and hence the risk of xruns and other detrimental behaviour.

The cause seems to be that loading any zyngui module triggers __init__.py which loads all the modules. I wonder if such an invasive initialisation is required?

**Comment by riban-bw** [comment]
Actually I am still struggling to work around this. Attempts to configure UI to control a library are also applied when webconf loads the same code. I will find a work-around but this is quite a significant issue.

**Comment by jofemodo** [comment]
This is exactly what it's imported from webconf:

```
sys.path.append(os.environ.get('ZYNTHIAN_UI_DIR'))
import zynconf
import zyngine.zynthian_lv2 as zynthian_lv2
from zyngine.zynthian_engine_pianoteq import *
from zyngine.zynthian_midi_filter import MidiFilterScript
from zyngui.zynthian_gui_engine import *
from zyngui.zynthian_gui_keybinding import zynthian_gui_keybinding
from zyngine.zynthian_engine_mixer import *

```

If you execute this from python shell:

```
root@zynthian:/zynthian/zynthian-ui# python3
Python 3.7.3 (default, Jan 22 2021, 20:04:44) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> import os
>>> sys.path.append(os.environ.get('ZYNTHIAN_UI_DIR'))
>>> 
>>> import zynconf
>>> import zyngine.zynthian_lv2 as zynthian_lv2
lilv_world_load_bundle(): warning: Replacing version 2071.7 of <http://gareus.org/oss/lv2/b_whirl#extended> from <file:///zynthian/zynthian-plugins/lv2/b_whirl.lv2/>
lilv_world_load_bundle(): note: New version 2071.11 found in <file:///zynthian/zynthian-plugins/lv2/b_whirl/>
lilv_world_load_bundle(): warning: Replacing version 2071.7 of <http://gareus.org/oss/lv2/b_whirl#simple> from <file:///zynthian/zynthian-plugins/lv2/b_whirl.lv2/>
lilv_world_load_bundle(): note: New version 2071.11 found in <file:///zynthian/zynthian-plugins/lv2/b_whirl/>
lilv_world_load_bundle(): warning: Ignoring bundle <file:///zynthian/zynthian-plugins/lv2/b_synth.lv2/>
lilv_world_load_bundle(): note: Newer version of <http://gareus.org/oss/lv2/b_synth> loaded from <file:///zynthian/zynthian-plugins/lv2/b_synth/>
>>> from zyngine.zynthian_engine_pianoteq import *
>>> from zyngine.zynthian_midi_filter import MidiFilterScript
>>> from zyngui.zynthian_gui_engine import *
New instance of zynseq
>>> from zyngui.zynthian_gui_keybinding import zynthian_gui_keybinding
>>> from zyngine.zynthian_engine_mixer import *
>>> 

```
A new instance of zynseq is created. By importing zynthian UI libraries, no instances should be created. Only the needed stuff for accessing the "static" functions needed by webconf. All instances must be created explicitly from zynthian-ui main object.

**Comment by jofemodo** [comment]
OK! I have created a development branch for addressing this.
I've removed from the zyngui module's "__init__.py" all the unneeded files and refactored the import sections on every class, so webconf can load in a fine-grained way what it's needed and only what's needed.
I haven't do the same with the "zyngine" module because we have to load every engine class from webconf to access "zynapi" static functions.
With this, the problem should be fixed. Of course, when a proper live-API is implemented, we could avoid loading all this stuff from webconf and use the live-API instead.
